### PR TITLE
new cli noun `checkpoint`, tap warns of deprecation

### PR DESCRIPTION
**Pull request by Aylr**
Changes proposed in this pull request: - - -

- new checkpoint noun in cli
- cli `tap` now warns it will be deprecated in the next major release

After submitting your PR, CI checks will run and @tiny-tim-bot will check for your CLA signature.

For nontrivial changes, we will conduct both a design review and a code review. Please tag a core team member for code review, or leave blank and we will find someone for you!

- [ ] Design Review (@jcampbell)
- [x] Code Review

Thank you for submitting!

### CODE REVIEW

File: great_expectations/cli/checkpoint.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
+import os
```
Comment (Aylr) [inline great_expectations/cli/checkpoint.py:1]: This was the original tap file renamed to checkpoint. It is not new work.

File: great_expectations/cli/tap.py [hunk 2]
```diff
@@ -36,70 +20,11 @@ def tap():
...
-
-
-def _write_tap_file_to_disk(batch_kwargs, context_directory, suite, tap_filename):
-    tap_file_path = os.path.abspath(os.path.join(context_directory, "..", tap_filename))
-
-    template = _load_template().format(
-        tap_filename, context_directory, suite.expectation_suite_name, batch_kwargs
+    """Create a new tap file for easy deployments."""
+    _checkpoint_new(
+        suite, tap_filename, directory, usage_event="cli.tap.new", datasource=datasource
```
Comment (Aylr) [inline great_expectations/cli/tap.py:29]: Tap simply points to the new checkpoint command after a warning.

**Review by ayirplm — APPROVED** [review]
LGTM

**Comment by Aylr** [comment]
Merging. Required checks passed: https://travis-ci.org/github/great-expectations/great_expectations/builds/678820778

**Pull request merged**